### ISSUE TITLE
Visualizing costmaps in the opengl window

### ISSUE THREAD

**Issue by gaya-** [cram-code/cram_physics#18]
Right now we have global variables for the Z and the orientation. See b8ae8b462f13aa76dacc80d97ffb4cf7aef19c55

Would be nice to be able to specify those during costmap generation not as a global variable for all the costmaps.

**Comment by fairlight1337** [comment]
Was this change done out of practicability reasons? Looks like the height was there already (`z`) :). Maybe we can extend that?

**Comment by gaya-** [comment]
It is there in the low-level API but gets lost in the high-level somewhere. I didn't look much into this, just opened this issue not to forget about it.

**Comment by fairlight1337** [comment]
Hm, thinking about it again, doesn't the OpenGL visualization already show the costmaps on the floor or a table, depending on the z-value generator within the location generators? Such as `make-semantic-map-height-function` in `semantic_map_costmap` (https://github.com/cram-code/cram_highlevel/blob/master/semantic_map_costmap/src/cost-functions.lisp#L338)?

My OpenGL visualization draws locations for robot standing poses on the floor, and for locations on table, it draws them on the table. The height is decided based on the generators registered for those location designators through the respective `(<- (desig-costmap ...))` predicates.

What exactly seems to be the problem then?

**Comment by gaya-** [comment]
No, unfortunately, it is always generated on the floor. The reason why it seems to be on the table when we request locations for objects is that the values of the costmap are relatively high (if the distribution is uniform it's 1.0, otherwise it's usually still close to 1.0), and the table height is about 0.8, so it seems like it's on the table. But if the distribution would not have such high values you wouldn't be able to see some parts of it because they would be inside the table. So, basically, the reds of the robot pose distributions are on the same height as the reds of the object distributions. So it isn't connected to the height generators. Which is exactly what I would prefer to have.

The point where the visualization is created is here: https://github.com/cram-code/cram_physics/blob/master/bullet_reasoning_designators/src/location-designator-integration.lisp#L61 and there the Z value is left on the default of 0.0.

**Comment by fairlight1337** [comment]
Oh, I see. Okay then, haven't had any issues with that.
